apps/kol-amelamdim/pages/login/index.tsx

```diff
@@ -18,69 +18,81 @@ const Login = () => {
   const [email, setEmail] = useState<string>('');
   const [password, setPassword] = useState<string>('');
   const [error, setError] = useState<string>('');
+  const [loading, setLoading] = useState<boolean>(false);
+
   const { setAuthenticated } = useContext(AuthContext);
   const router = useRouter();
 
   const handleSubmit = async (e) => {
     e.preventDefault();
     if (validator.isEmail(email)) {
+      setLoading(true);
       try {
         const { data } = await axios.post('/api/login', { email, password });
         if (data.success) {
           setAuthenticated(true);
-          router.push('/');
+          await router.push('/');
+          setLoading(false);
         }
       } catch (error) {
+        setLoading(false);
         setError(error.response.data.message.heb);
       }
     } else {
+      setLoading(false);
       setError(API_ERRORS.invalidEmailError.message.heb);
     }
   };
-
+  //Todo: handle loading and errors
   return (
     <StyledPageContainer>
-      <form onSubmit={handleSubmit}>
-        <Grid container direction={'column'}>
-          <Typography variant="h3" component="h2" sx={{ mt: 2 }}>
-            התחברות
-          </Typography>
-          <TextField
-            sx={{ mt: 2 }}
-            required
-            id="outlined-required"
-            label="אימייל"
-            value={email}
-            onChange={(e) => setEmail(e.target.value)}
-            error={!!error}
-          />
-          <TextField
-            sx={{ mt: 2 }}
-            value={password}
-            label="סיסמא"
-            type="password"
-            required
-            autoComplete="current-password"
-            onChange={(e) => setPassword(e.target.value)}
-            error={!!error}
-          />
-          <Button
-            sx={{ mt: 2 }}
-            variant="contained"
-            type="submit"
-            disabled={!email || !password}
-          >
-            התחבר
-          </Button>
-        </Grid>
-      </form>
-      <Grid container sx={{ mt: 2 }}>
-        <Typography component="h4">אין לכם משתמש?&nbsp;</Typography>
-        <NextLink href="/register" passHref>
-          <MUILink>לחצו להרשמה</MUILink>
-        </NextLink>
-      </Grid>
-      {error && <FormError>{error}</FormError>}
+      {loading ? (
+        <div>טוען....</div>
```

Comment:
I didn't like the experience. When use clicking we are "removing" everything from the screen and showing him that "loading" text and if it's failing we are going back to the old state.

I think loading indicators should be in the same screen *with* the same elements

Reply:
yea i have to do that, this is just a place holder for now..